### DISCUSSION TITLE
关于固件源码和SDK代码的疑问

### DISCUSSION THREAD

**Discussion by GreenShadeZhang** [peng-zhihui/ElectronBot#184] [Q&A]
本来最近在看固件和SDK源码的时候遇到一些问题，为什么固件源码和sdk源码里这里需要循环四次呢？
四次有什么讲究嘛，有没有来帮忙讲讲思路的。
![QQ截图20220524155616](https://user-images.githubusercontent.com/24240675/169980304-7f5fff73-a505-4f8b-ac64-c37be992d460.png)

**Comment by GreenShadeZhang** [comment]
自问自答，原来4次刚好把图片的数据传输完成，（84x512+192）x4=172800 和240x240x3相同。

**Reply by VincentYoungHao** [reply]
我觉得是 60*240*3*4 = 172800 较合理，程序后面就是这样干的。至于如何得出 （84x512+192）x4=172800 的我没有研究，能说下你是怎么推出得吗？

**Reply by VincentYoungHao** [reply]
60`*`240`*`3`*`4 `=` 172800

**Comment by zlq12** [comment]
请问一下，固件和sdk的开发这块有没有什么好的教程，视频或者书籍，本人新手

**Reply by GreenShadeZhang** [reply]
> 请问一下，固件和sdk的开发这块有没有什么好的教程，视频或者书籍，本人新手

网上买套stm32的硬件，跟着教程学就行了，和学51一样。

**Comment by Kitem** [comment]
您好，我是焕挺，您的邮件已经收到，我会尽快回复！

**Comment by Kitem** [comment]
您好，我是焕挺，您的邮件已经收到，我会尽快回复！
